src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/mha_single_token.cpp

```diff
@@ -175,7 +175,15 @@ void mha_single_token_kernel(const ov::intel_cpu::PlainTensor& query,
         // apply attention mask & sofmax
         auto ncausal = auto_causal ? (kv_len - q_len + pq + 1) : kv_len;
         float* alibi_ptr = alibi_mask ? &alibi_mask.at<float>({b, h, pq, 0}, true) : nullptr;
-        float* attn_mask_ptr = attention_mask ? &attention_mask.at<float>({b, h, pq, 0}, true) : nullptr;
+        uint8_t* attn_mask_ptr = nullptr;
+        auto attn_mask_prec = attention_mask.get_precision();
+        if (attention_mask) {
+            if (attn_mask_prec == ov::element::f32) {
+                attn_mask_ptr = reinterpret_cast<uint8_t*>(&attention_mask.at<float>({b, h, 0, 0}, true));
+            } else {
+                attn_mask_ptr = reinterpret_cast<uint8_t*>(&attention_mask.at<bfloat16>({b, h, 0, 0}, true));
+            }
+        }
```

Comment:
Is the element type of attn_mask aligned with that of querry? If so can we use template parameter?

Reply:
Maybe, it relies on the rtPrecision.

Reply:
Fix in a518f574611468d8a444da0e635480f2572bb316